docker/linksets/mb2cc/linksets_builder.py

```diff
@@ -87,28 +109,29 @@ def create_geometry_indexes():
     '''
     logging.info("Creating Geometry Indexes")
     create_geometry_indexes_sql = """
-    CREATE INDEX mb_geom_3577_gix ON public.\"mb:mb\" USING GIST (geom_3577);
+    CREATE INDEX mb_geom_3577_gix ON public.\"mb_mb\" USING GIST (geom_3577);
     CREATE INDEX cc_geom_3577_gix ON public.\"ahgfcontractedcatchment\" USING GIST (geom_3577);
     """
     run_command(["psql", "--host", "postgis", "--user",
                  "postgres", "-d", "mydb", "-c", create_geometry_indexes_sql])
 
+
 def create_intersections():
     logging.info("Calculating intersecting mb and cc")
     create_intersection_sql = """
-CREATE MATERIALIZED VIEW mbintersectcc_mv AS
-SELECT mb.mb_code_2016, ca.hydroid, ST_Intersection(mb.geom_3577, ca.geom_3577) as i
-FROM public.\"ahgfcontractedcatchment\" as ca
-INNER JOIN public.\"mb:mb\" as mb ON mb.geom_3577 && ca.geom_3577 -- the && specifies an indexed bounding box lookup
-WHERE ST_IsValid(ca.geom_3577) AND ST_IsValid(mb.geom_3577) AND ST_Intersects(mb.geom_3577, ca.geom_3577)
-ORDER BY mb.mb_code_2016 ASC;
-CREATE MATERIALIZED VIEW ccintersectmb_mv AS
-SELECT ca.hydroid, mb.mb_code_2016, ST_Intersection(ca.geom_3577, mb.geom_3577) as i
-FROM public.\"mb:mb\" as mb
-INNER JOIN public.\"ahgfcontractedcatchment\" as ca ON ca.geom_3577 && mb.geom_3577 -- the && specifies an indexed bounding box lookup
-WHERE ST_IsValid(ca.geom_3577) AND ST_IsValid(mb.geom_3577) AND ST_Intersects(ca.geom_3577, mb.geom_3577)
-ORDER BY ca.hydroid ASC;
-"""
+    CREATE MATERIALIZED VIEW mbintersectcc_mv AS
+    SELECT mb.mb_code_20, ca.hydroid, ST_Intersection(mb.geom_3577, ca.geom_3577) as i
```

Comment:
Why is `mb_code_2016` changed to `mb_code_20`?

Where is the `mb_code_20` name coming from?

Reply:
Because now tasgs mb data is loaded from a cached representation on s3 namely a shape file and in that the field name has been truncated. In the next evolution of this code at https://github.com/CSIRO-enviro-informatics/loci-cache-scripts/tree/generalize_database_load the field names have been factored out to a large extent.

Reply:
Ok, but why does the cached representation in S3 have that field name truncated? That seems like a bug. `mb_code_2016` could be short for `mb_code_2011` or `mb_code_2016`, both are valid but different property names in the asgs dataset.

Reply:
Your right that both `mb_code_2011` and `mb_code_2016` could be truncated to `mb_code_20` however this code specifically is for 2016 mb see line 12 of docker/linksets/mb2cc/linksets_builder.py and all of docker/linksets/mb2cc/preload_asgs_wfs.py. As, I mentioned in the next batch of changes, `mb_code_20` is being factored out to a specific identifier parameter to work for other linksets (including 2011 mb).  The truncation is because shape files have a 10 character field limit see https://gis.stackexchange.com/questions/15784/bypassing-10-character-limit-of-field-name-in-shapefiles. 

For the generic version of this code it is true that two cached copies one of 2011 and one of 2016 mb could both share the field name `mb_code_20` futhermore other source datasets could share identifier fields like `id`. Collisions across different cached `from` or `to` datasets won't matter in this context because they are never used concurrently, that is linkset build deals with one linkset at a time.

Reply:
Actually do you mean within the asgs mb dataset? Because that would be an issue. I'll have a look.

Reply:
Ok I've had a look at the source WFS data for the 2016 mb, that is what is being used to generate the cache (see the `.env` file) e.g `https://geo.abs.gov.au/arcgis/services/ASGS2016/MB/MapServer/WFSServer?service=wfs&version=2.0.0&request=GetFeature&typeName=MB:MB&Filter=%3Cogc:Filter%3E%3Cogc:PropertyIsEqualTo%3E%3Cogc:PropertyName%3EMB:MB_CODE_2016%3C/ogc:PropertyName%3E%3Cogc:Literal%3E50055290000%3C/ogc:Literal%3E%3C/ogc:PropertyIsEqualTo%3E%3C/ogc:Filter%3E` and I can confirm there is no 2011 property in there just `MB_CODE_2016`

However you make a good point that these collisions could be an issue *within* a dataset if two fields get truncated to the same length. I'll have a look at `ogr2ogr` parameters and see what we might be able to do to avoid that.

Reply:
Looking into the `ogr2ogr` documentation and the related behaviour documented at https://gdal.org/drivers/vector/shapefile.html name collisions within a dataset are handled by the shapefile driver.

The ongoing refactor factors out the `mb_code_20` identifier altogether and expects the "user" or implementer of specific linkset build processes to parameterize it see:
https://github.com/CSIRO-enviro-informatics/loci-cache-scripts/compare/small_load_enabled...generalize_database_load#diff-a0b654b8a8a4d1609ac42bbb14e505e7R160

In this case the user will have to understand what the unique and appropriate identifier is in "from" or "to" datasource they build. 

Possibly we could document this but I don't think it is currently a bug or likely to be in the near future.

Reply:
Ok. All good. Thanks for looking into it. Yes, I thought the source data for ASGS 2016 did have MB_CODE_2011 properties. But you confirmed it doesn't, so thats fine.